util/io/game_state.py

```diff
@@ -14,7 +14,9 @@ def __init__(self, packet: GameTickPacket, previous_game_state, index: int):
         for car in packet.game_cars:
             team = Team.idToTeam(car.team)
             self.car_list.append(CarData(car, team))
-        self.previousGameState = previous_game_state
+        self.previous_game_state = previous_game_state
+        if self.previous_game_state is not None:
+            self.previous_game_state.previous_game_state = None

```

Comment:
just removing n-2 gamestates because we don't need them, and it takes a huge amount of data for no reason